### PR TITLE
bugfix:  fix fastjson serialization of time data types

### LINKED ISSUE #4443: AT模式下，fastJson序列化时，对于Mysql DATE相关类型全部序列化成了字符串，导致镜像对比出错
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
针对issue#4402 对AT模式下的mysql db type做了一轮测试 https://github.com/seata/seata/pull/4402
2) AT模式下，fastJson序列化时，对于Mysql DATE相关类型全部序列化成了字符串，导致二阶段回滚数据校验时镜像对比出错。DataCompareUtils类中convertType()方法目前只对TIMESTAMP类型做了特殊处理，应该要加上DATE和TIME类型的处理
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)
在Mysql5及Mysql8下都可以复现该问题，可以使用https://github.com/seata/seata/pull/4402的测试用例进行测试

### Ⅵ. Environment:

- JDK version :jdk8
- Seata version:  1.4.2
- OS :windows
- Others:mysql: 5.7.24或8.0.12

### PR DESCRIPTION
**Pull request by xjlgod**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
Detail is in [issue 4443](https://github.com/seata/seata/issues/4443)

### Ⅱ. Does this pull request fix one issue?

fixes #4443

### CODE REVIEW

**Review by slievrly — APPROVED** [review]
LGTM

**Comment by slievrly** [comment]
@xjlgod add commit log in change/1.5.0:
https://github.com/seata/seata/blob/develop/changes/1.5.0.md
https://github.com/seata/seata/blob/develop/changes/en-us/1.5.0.md

**Comment by xjlgod** [comment]
> @xjlgod add commit log in change/1.5.0: https://github.com/seata/seata/blob/develop/changes/1.5.0.md https://github.com/seata/seata/blob/develop/changes/en-us/1.5.0.md

I have done it.

**Comment by xjlgod** [comment]
> https://github.com/seata/seata/pull/4505#issuecomment-1087718470
I have done it.

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java [hunk 1]
```diff
@@ -91,6 +93,20 @@ private DataCompareUtils() {
...
+        if (f1Type == Types.DATE && f1.getValue().getClass().equals(String.class)) {
+            String[] strings = f1.getValue().toString().split(" ");
+            f1.setValue(Date.valueOf(strings[0]));
+        }
+        if (f0Type == Types.TIME && f0.getValue().getClass().equals(String.class)) {
+            f0.setValue(Time.valueOf(f0.getValue().toString()));
+        }
+        if (f1Type == Types.TIME && f1.getValue().getClass().equals(String.class)) {
+            f1.setValue(Time.valueOf(f1.getValue().toString()));
+        }
```
Comment (pengten) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java:109]: date处理了类似 2022-04-08 12:23:33 的情况，time不需要处理吗？

Comment (xjlgod) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java:109]: time本身就是12:23:33这种类型，不需要做处理。主要是fastJson序列化出来的字符串需要转成time类型来比较。

Comment (wskiawv) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java:109]: 数据库是datetime类型的如何处理？

Comment (xjlgod) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataCompareUtils.java:109]: > https://github.com/seata/seata/pull/4505#discussion_r847939475
解决的issue里没涉及对datetime类型错误的描述，datetime是相当于TIMESTAMP进行处理的

**Comment by a364176773** [comment]
你的分支过老了,我无法帮你合并最新的develop代码
image
请尽快合并最新seata develop分支代码并推送到此分支,以便我们进行合并pr

**Pull request merged**